src/tactic/unify_equations.lean

```diff
@@ -0,0 +1,407 @@
+/-
+Copyright (c) 2020 Jannis Limperg. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Jannis Limperg
+-/
+import data.nat.basic
+
+/-!
+# The `unify_equations` tactic
+
+This module defines `unify_equations`, a first-order unification tactic that
+unifies one or more equations in the context. It implements McBride's Qnify
+algorithm [^qnify].
+
+The tactic takes as input some equations which it simplifies one after the
+other. Each equation is simplified by applying one of several possible
+unification steps. Each such step may output other (simpler) equations which are
+unified recursively until no unification step applies any more. See
+`tactic.interactive.unify_equations` for an example and an explanation of the
+different steps.
+
+[^qnify]: Conor McBride. Inverting inductively defined relations in LEGO.
+In E. Giménez and C. Paulin-Mohring (eds.), Types for Proofs and Programs 1996.
```

Comment:
This can be added to `docs/references.bib`, see e.g. the citations [here](https://github.com/leanprover-community/mathlib/blob/a18f142d831860e6dd0efdf6117a91700764eee7/src/set_theory/pgame.lean#L97).